### PR TITLE
[MCKIN-7601] Rename APIs for Native UI

### PR DESCRIPTION
**Pull request by viadanna**
1. Rename get_configuration -> student_view_data
2. Rename get_user_state -> student_view_user_state

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @viadanna! I've created [OSPR-2454](https://openedx.atlassian.net/browse/OSPR-2454) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



If you like, you can add yourself to the [AUTHORS](https://github.com/edx-solutions/xblock-drag-and-drop-v2/blob/master/AUTHORS)      file for this repo, though that isn't required.     Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

**Comment by viadanna** [comment]
Missing some commits, I'll reopen when done.

File: Native_API.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
+Native API Documentation
```
[the remark this answers is not in the document]
Comment (viadanna) [reply Native_API.md]: 👍

File: drag_and_drop_v2/drag_and_drop_v2.py [hunk 2]
```diff
@@ -308,6 +310,10 @@ def items_without_answers():
             return items
 
         return {
+            # "block_id": unicode(self.scope_ids.usage_id),
```
Comment (xitij2000) [inline drag_and_drop_v2/drag_and_drop_v2.py]: Please uncomment this.

File: setup.py [hunk 3]
```diff
@@ -23,7 +23,7 @@ def package_data(pkg, root_list):
 
 setup(
     name='xblock-drag-and-drop-v2',
-    version='2.1.6',
+    version='2.1.7',
```
Comment (xitij2000) [inline setup.py]: @viadanna  Bump this to 2.2.0

File: tests/unit/test_advanced.py [hunk 4]
```diff
@@ -63,7 +63,7 @@ def expected_configuration(cls):
         return json.loads(loader.load_unicode('data/{}/config_out.json'.format(cls.FOLDER)))
 
     def test_get_configuration(self):
```
Comment (xitij2000) [inline tests/unit/test_advanced.py]: Rename the tests as well. Sorry for not noticing before.

File: tests/unit/test_basics.py [hunk 5]
```diff
@@ -50,13 +50,17 @@ def test_template_contents(self):
 
     def test_get_configuration(self):
```
Comment (xitij2000) [inline tests/unit/test_basics.py]: Same, rename the test.

**Comment by xitij2000** [comment]
@viadanna Could you please address the failing tests, and then update the mckinsey-master branch accordingly.

**Comment by viadanna** [comment]
Block ids aren't stable, I had to work around that in tests.

**Comment by xitij2000** [comment]
@sanfordstudent  Could you have a look at this PR? It is ready for review now.

**Comment by pomegranited** [comment]
@mduboseedx Since @sanfordstudent is leaving edX soon, who should we ping for upstream reviews in the future?

**Comment by mduboseedx** [comment]
@pomegranited Our Educator team should be handling drag n drop changes.

@scottrish Do you know who can do upstream reviews for drag n drop PRs?

**Comment by pomegranited** [comment]
Thanks for sorting out ownership of this XBlock @mduboseedx !
@scottrish McKinsey Academy are working on several improvements to the functionality which affect the GUI for DnDv2, and OpenCraft will be helping to package them up for upstream review.  What's the best way for us to discuss these changes with you?  Are individual PRs enough, or would you like a separate OSPR or email thread to review the full plan? Thank you! CC @xitij2000

**Comment by edx-webhook** [comment]
Thanks for the pull request, @viadanna! I've created [OSPR-2533](https://openedx.atlassian.net/browse/OSPR-2533) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



If you like, you can add yourself to the [AUTHORS](https://github.com/edx-solutions/xblock-drag-and-drop-v2/blob/master/AUTHORS)      file for this repo, though that isn't required.     Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

**Comment by xitij2000** [comment]
@sstack22 Could you have a look at this please? This is ready for review. Sorry I forgot to ping earlier.

**Comment by xitij2000** [comment]
@sstack22 Could you have a look at this please? This is ready for review. Sorry I forgot to ping earlier.

**Comment by sstack22** [comment]
@xitij2000 - thanks for the ping! From reading the first comment on this PR, it's not immediately clear to me what specifically needs to be reviewed from a product perspective. It seems like these are changes to the native app - are there specific learner facing changes? Or is it really just renaming the APIs?

**Comment by xitij2000** [comment]
@sstack22 It just renames the API to match the standard names for mobile API since this block pre-dates them. These would only affect a native app using these APIs.

**Comment by marcotuts** [comment]
No concerns on my end. I may ask both @ormsbee to take a look here (feel free to hand off to nimisha if her mobile context simplifies review.) Also, will have mobile team review after initial architecture review. Thanks!

**Comment by ormsbee** [comment]
Please rebase and I can merge it.

**Comment by viadanna** [comment]
Done, thanks @ormsbee

**Pull request merged**